File: C/09_Math2/BOJ_05_4948_베르트랑공준.c
#include <stdio.h>
#include <math.h>
int boolean[246913];
main() {
	boolean[1] = 1;
	for (int i = 2; i <= (int)sqrt(246912); i++) {
		if (boolean[i]) continue;
		for (int j = i * i; j <= 246912; j += i) {
			boolean[j] = 1;
		}
	}
	while (1) {
		int n, cnt;
		scanf("%d", &n);
		if (n == 0) break;
		cnt = 0;
		for (int i = n + 1; i <= 2 * n; i++) {
			if (!boolean[i]) cnt++;
		}
		printf("%d\n", cnt);
	}
}

File: JAVA/09_Math2/BOJ_05_4948_베르트랑공준.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		boolean[] tst = new boolean[246913];
		tst[1] = true;
		for (int i = 2; i <= (int) Math.sqrt(246912); i++) {
			if (tst[i]) continue;
			for (int j = i * i; j <= 246912; j += i) {
				tst[j] = true;
			}
		}
		StringBuilder sb = new StringBuilder();
		while (true) {
			int n = Integer.parseInt(br.readLine());
			if (n == 0) break;
			int cnt = 0;
			for (int i = n + 1; i <= 2 * n; i++) {
				if (!tst[i])
					cnt++;
			}
			sb.append(cnt).append("\n");
		}
		System.out.print(sb);
	}
}